packages/ember-data/lib/system/model/states.js

```diff
@@ -343,7 +345,7 @@ var DirtyState = {
 
     rolledBack: function(record) {
       get(record, 'errors').clear();
-      record.triggerLater('ready');
+      record.transitionTo('loaded.saved');
```

Comment:
> or the behavior of rolling back an invalid record is currently not supported

This is indeed the current behavior: rolling back a new record which is marked as invalid puts the record in the `loaded.created.uncommitted` state. Is this the intended behavior? There are currently no tests for this case in [`rollback-test.js`](https://github.com/emberjs/data/blob/master/packages/ember-data/tests/unit/model/rollback-test.js) so it needs to be discussed what should be the intended behavior here. Sorry for hijacking this PR for this issue, I will gladly open a new one if this is desired ...